models/classes/LtiProvider/LtiProvider.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoLti\models\classes\LtiProvider;
+
+/**
+ * LTI provider business object.
+ */
+class LtiProvider implements \JsonSerializable
+{
+    /** @var string */
+    private $uri;
+
+    /** @var string */
+    private $label;
+
+    /** @var string */
+    private $key;
+
+    /** @var string */
+    private $secret;
+
+    /** @var string */
+    private $callbackUrl;
+
+    /**
+     * LtiProvider constructor.
+     *
+     * @param string $uri
+     * @param string $label
+     * @param string $key
+     * @param string $secret
+     * @param string $callbackUrl
+     */
+    public function __construct($uri = '', $label = '', $key = '', $secret = '', $callbackUrl = '')
+    {
+        $this->uri = $uri;
+        $this->label = $label;
+        $this->key = $key;
+        $this->secret = $secret;
+        $this->callbackUrl = $callbackUrl;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUri()
+    {
+        return $this->uri;
+    }
+
+    /**
+     * @param string $uri
+     *
+     * @return LtiProvider
+     */
+    public function setUri($uri)
```

Comment:
Are we going to use these setters? Shouldn't be immutable?

Reply:
Made it immutable.